### PR TITLE
Explore Regions Additions

### PR DESCRIPTION
**Pull request by citlali-trigos-raczkowski**
list of changes
- removing the mediterranean page -->couldn't figure out how to do this page without there being a lot of repetition with the "explore regions" page, so instead focused on bulking up the explore regions page
- a lot of changes are from formatting the code
- introduce new package `millify` which formats numbers to readable string: `987987 --> 987,987`. You can install this package via `pip3 install millify`
- added a line including the breakdown of deaths by men, women, and children categories
- Removed some X / Y labels (ex: "Year" text under list 2018, 2019, ....)
- Lot of cleanup: typos, spaces, titles, capitalizations, etc

I'll leave some comments in the code to point out some changes

### CODE REVIEW

File: src/Home.py [hunk 1]
```diff
@@ -51,114 +53,124 @@ def get_df():
...
+    margin={"r": 0, "t": 0, "l": 0, "b": 0},
+)
+
+plot = px.scatter_geo(migrantdf, lat='lat', lon='lon',
+                      hover_name='Migration route',
+                      hover_data={'Cause of Death Abbreviation': True,
+                                  'Total Number of Dead and Missing': True,
+                                  'Incident year': True,
+                                  'lon': False,
+                                  'lat': False
```
Comment (citlali-trigos-raczkowski) [inline src/Home.py:149]: Adding these 2 lines removed the X Y coordinates showing up in the hover data on the map!

File: src/pages/Explore_Regions.py [hunk 2]
```diff
@@ -39,45 +41,50 @@ def plot_deaths_season(m_route, df):
     dft = df[df['Migration route'] == m_route]
 
     fig = px.line(dft, x='Incident year', y='count',
-                  color='Season', title=f"Total deaths per Season in {m_route}")
+                  color='Season', title=f"Total deaths per season in {m_route}")
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:44]: moving to lowercase, looks like most other titles are using this format

File: src/pages/Explore_Regions.py [hunk 3]
```diff
@@ -39,45 +41,50 @@ def plot_deaths_season(m_route, df):
...
         dft = df_wout_cause[df_wout_cause['Migration route'] == m_route]
-    fig = px.line(dft, x='date', y=['Total Number of Dead and Missing','Number of Females', 'Number of Males', 'Number of Children'], title=f"Plot of deaths by month along the {m_route} route.")
+    fig = px.line(dft, x='date', y=['Total Number of Dead and Missing', 'Number of Females',
+                  'Number of Males', 'Number of Children'], title=f"Plot of deaths by month along the {m_route} route.")
     fig.update_layout({
         'plot_bgcolor': 'rgba(0, 0, 0, 0)',
         'paper_bgcolor': 'rgba(0, 0, 0, 0)',
-        'xaxis_title':"Date", 
-        'yaxis_title':"Number of Deaths",
+        'xaxis_title': "",
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:65]: cleaner and i think not necessary

File: src/pages/Explore_Regions.py [hunk 4]
```diff
@@ -119,54 +127,101 @@ def plot_comp(m_route, cause, df):
...
-"Migrations that cross borders and routes include entire families with children. Those routes are so dangerous, and even more for infants." + "On the other hand, others manage to survive the journey, but the fact that they do does not mean that it was without great difficulty and that many others did not.")
+st.markdown("In this page we welcome you to explore the data one region at a time, using the filters in the left side menu. Try selecting a new region to check how the data visualization changes. " +
+            "Migrations that cross borders and routes often include families with children. While extremely dangerous, they are even moreso for infants." + " For those that do survive the journey, it is not without great difficulty and loss.")
+
+st.markdown("The following is a table of all included migration routes paired with the total number of recorded dead and missing people from that route, since 2014.")
+dff = df.groupby(['Migration route'])[
+    ['Total Number of Dead and Missing']].sum().sort_values('Total Number of Dead and Missing', ascending=False).reset_index()
+st.dataframe(dff)
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:136]: Table of columns <migration route> and <total number of dead/missing> sorted from highest to lowest, available for the user to explore before selecting a region to look at

File: src/pages/Explore_Regions.py [hunk 5]
```diff
@@ -119,54 +127,101 @@ def plot_comp(m_route, cause, df):
...
+
 st.sidebar.write("Data Filters")
+
 route_input = [st.sidebar.selectbox(
     'Migration Route', df['Migration route'].unique().tolist())]
+st.header('Exploring the '+route_input[0])
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:142]: Added in this header and subheaders/markdown throughout to explain what it is we're showing

File: src/pages/Explore_Regions.py [hunk 6]
```diff
@@ -119,54 +127,101 @@ def plot_comp(m_route, cause, df):
...
+    total_women_dead_missing = prettify(df_d_s['Number of Females'].iloc[0])
+    total_men_dead_missing = prettify(df_d_s['Number of Males'].iloc[0])
+    total_children_dead_missing = prettify(
+        df_d_s['Number of Children'].iloc[0])
+
+    # introduction
+    st.markdown("For the " + route_input[0] +
+                " migration route, there are a total of  " + total_dead_missing +
+                " recorded deaths in the data since 2014. This includes " + total_women_dead_missing +
+                " women, " + total_men_dead_missing + " men, and " + total_children_dead_missing + " children.")
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:167]: Intro to the region includes breakdown of dead/missing by statistics on children/women/men

File: src/pages/Explore_Regions.py [hunk 7]
```diff
@@ -119,54 +127,101 @@ def plot_comp(m_route, cause, df):
...
+
+    # map of the selection region
+    # TODO: there's some data that shows up that shouldn't be here..
+    st.map(mdf)
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py]: BIG todo.. I'm seeing data show up that shouldn't be here... You can see if on the main branch if you look at the US-MX migration route, there's dots on the other side of the world

Comment (citlali-trigos-raczkowski) [reply src/pages/Explore_Regions.py]: (assuming this is a mistake)

File: src/pages/Explore_Regions.py [hunk 8]
```diff
@@ -119,54 +127,101 @@ def plot_comp(m_route, cause, df):
...
+    # Country of origin
+    mdf['Number of People'] = mdf.loc[:, [
+        'Total Number of Dead and Missing', 'Number of Survivors']].sum(axis=1)
+    total_migrants = mdf['Number of People'].sum()
+    mdf['Percentage of People from Country of Origin'] = mdf['Number of People']/total_migrants
+    dforigin = mdf.groupby(['Country of Origin'])[['Number of People', 'Percentage of People from Country of Origin']].sum(
+    ).sort_values('Number of People', ascending=False).reset_index()
+
+    st.markdown("The following table lists the known countries of origin for migrants traveling along this route. For each country, we see the total number of people from the country and the percentage of migrants from the country.")
+    st.dataframe(dforigin)
```
Comment (citlali-trigos-raczkowski) [inline src/pages/Explore_Regions.py:178]: Country of origin data! displayed as a table. Might be interesting to display as a heat map or something

File: src/Home.py (hunk 1 above)
Comment (srodriguezb) [reply src/Home.py:149]: Great

File: src/pages/Explore_Regions.py (hunk 7 above)
Comment (srodriguezb) [reply src/pages/Explore_Regions.py]: So, that map is for a test or are we putting another map in the Explore regions? Also, if we are going to add a map, is better to use px.scatter_geo() than st.map()

**Review by srodriguezb — APPROVED** [review]
The only thing that confuses me is the new map

**Comment by citlali-trigos-raczkowski** [comment]
@srodriguezb 
Yeah I agree, I just feel like a lot of these routes people don't know where they are on the map, so it might be nice to have a "oh the X route is in this part of continent Y" would be helpful, but I agree that having another map doesn't really make sense, so I'll remove it

File: src/Home.py [hunk 9]
```diff
@@ -51,114 +53,139 @@ def get_df():
...
+st.dataframe(df_cause_of_death, height=200)
+
+col2, col3 = st.columns(2)
+with col2:
+    st.metric(f'Total number of Recorded Dead and Missing',
+              prettify(statsdf['Total Number of Dead and Missing'].sum()))
+with col3:
+    st.metric(f'Total Number of Recorded Survivors',
+              prettify(statsdf['Number of Survivors'].sum()))
+
```
Comment (citlali-trigos-raczkowski) [inline src/Home.py:105]: Added in these global statistics, with the ability to filter through via year:
Screen Shot 2022-10-22 at 11 12 06

**Pull request merged**